server/apps/immich/src/api-v1/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { UserEntity } from '@app/database/entities/user.entity';
+import { BadRequestException } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import * as bcrypt from 'bcrypt';
+import { AuthType } from '../../constants/jwt.constant';
+import { ImmichJwtService } from '../../modules/immich-jwt/immich-jwt.service';
+import { OAuthService } from '../oauth/oauth.service';
+import { IUserRepository, USER_REPOSITORY } from '../user/user-repository';
+import { AuthService } from './auth.service';
+import { SignUpDto } from './dto/sign-up.dto';
+import { LoginResponseDto } from './response-dto/login-response.dto';
+
+const fixtures = {
+  login: {
+    email: 'test@immich.com',
+    password: 'password',
+  },
+};
+
+const CLIENT_IP = '127.0.0.1';
+
+jest.mock('bcrypt');
+
+describe('AuthService', () => {
+  let sut: AuthService;
+  let userRepositoryMock: jest.Mocked<IUserRepository>;
+  let immichJwtServiceMock: jest.Mocked<ImmichJwtService>;
+  let oauthServiceMock: jest.Mocked<OAuthService>;
+  let compare: jest.Mock;
+
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  beforeEach(async () => {
+    jest.mock('bcrypt');
```

Comment:
This has already been mocked above. Is this necessary because of `jest.resetModules()`?

Reply:
I'll have to test it, but I don't think this is actually necessary.